collector/receiver/telemetryapireceiver/receiver.go

```diff
@@ -0,0 +1,199 @@
+// and put into a synchronous queue to be dispatched later.
+// Logging or printing besides the error cases below is not recommended if you have subscribed to
+// receive extension logs. Otherwise, logging here will cause Telemetry API to send new logs for
+// the printed lines which may create an infinite loop.
+func (r *telemetryAPIReceiver) httpHandler(w http.ResponseWriter, req *http.Request) {
+	body, err := io.ReadAll(req.Body)
+	if err != nil {
+		r.logger.Error("error reading body", zap.Error(err))
+		return
+	}
+
+	var slice []event
+	if err := json.Unmarshal(body, &slice); err != nil {
+		r.logger.Error("error unmarshalling body", zap.Error(err))
+		return
+	}
+
+	for _, el := range slice {
+		r.logger.Info(fmt.Sprintf("Event: %s", el.Type), zap.Any("event", el))
+		switch el.Type {
+		// Function initialization started.
+		case "platform.initStart":
+			r.logger.Info(fmt.Sprintf("Init start: %s", r.lastPlatformStartTime), zap.Any("event", el))
+			r.lastPlatformStartTime = el.Time
+		// Function initialization completed.
+		case "platform.initRuntimeDone":
+			r.logger.Info(fmt.Sprintf("Init end: %s", r.lastPlatformEndTime), zap.Any("event", el))
+			r.lastPlatformEndTime = el.Time
+		}
+		// TODO: add support for additional events, see https://docs.aws.amazon.com/lambda/latest/dg/telemetry-api.html
+		// A report of function initialization.
+		// case "platform.initReport":
+		// Function invocation started.
+		// case "platform.start":
+		// The runtime finished processing an event with either success or failure.
+		// case "platform.runtimeDone":
+		// A report of function invocation.
+		// case "platform.report":
+		// Runtime restore started (reserved for future use)
+		// case "platform.restoreStart":
+		// Runtime restore completed (reserved for future use)
+		// case "platform.restoreRuntimeDone":
+		// Report of runtime restore (reserved for future use)
+		// case "platform.restoreReport":
+		// The extension subscribed to the Telemetry API.
+		// case "platform.telemetrySubscription":
+		// Lambda dropped log entries.
+		// case "platform.logsDropped":
+	}
+	if len(r.lastPlatformStartTime) > 0 && len(r.lastPlatformEndTime) > 0 {
+		if td, err := r.createPlatformInitSpan(r.lastPlatformStartTime, r.lastPlatformEndTime); err == nil {
+			err := r.nextConsumer.ConsumeTraces(context.Background(), td)
+			if err == nil {
+				r.lastPlatformEndTime = ""
+				r.lastPlatformStartTime = ""
+			} else {
+				r.logger.Error("error receiving traces", zap.Error(err))
+			}
+		}
+	}
+
+	r.logger.Debug("logEvents received", zap.Int("count", len(slice)), zap.Int64("queue_length", r.queue.Len()))
+	slice = nil
+}
+
+func (r *telemetryAPIReceiver) createPlatformInitSpan(start, end string) (ptrace.Traces, error) {
+	traceData := ptrace.NewTraces()
+	rs := traceData.ResourceSpans().AppendEmpty()
+	ss := rs.ScopeSpans().AppendEmpty()
+	ss.Scope().SetName("receiver/telemetryapi")
+	span := ss.Spans().AppendEmpty()
+	span.SetSpanID(newSpanID())
```

Comment:
No TraceID?  It's not valid without a TraceID.

Reply:
Right, I was using this in collaboration w/ the coldspanprocessor which overrides the trace ID with the ID from the span it correlates the cold start to, but it should have a trace ID anyways, for the case where the coldstartprocessor isn't configured. I've added this now, it's generating a new trace ID, in the future this could be modified to pull the trace ID from the xray env variable.